### ISSUE TITLE
STK:  is there a non-hacky way to get STK_USE_SIMPLE_FIELDS defined?

### ISSUE THREAD

**Issue by bartgol** [trilinos/Trilinos#12547]
## Question
I am getting some errors in Albany (not master), which I ultimately tracked down to me re-declaring a field (that was already read in by stk_io) with the wrong template type. I was puzzled by this, since all our fields have just one template arg, the data type, and no tags. Digging a bit, I discovered that as the exo mesh is read in, vector fields are created with Cartesian tag, [here](https://github.com/trilinos/Trilinos/blob/58b8b785f4db5aa7a7fc960b87639287b9eb13ac/packages/stk/stk_io/stk_io/IossBridge.cpp#L637). However, that code could be avoided if meta data was using simple fields, see [this line](https://github.com/trilinos/Trilinos/blob/58b8b785f4db5aa7a7fc960b87639287b9eb13ac/packages/stk/stk_io/stk_io/IossBridge.cpp#L582) a little bit above.

Now, I can call `io_broker.use_simple_fields()` before loading the mesh, forcing the meta data to use simple fields, but I notice a lot of `STK_USE_SIMPLE_FIELDS` maros around stk, which would force meta data to _always_ be created with `m_use_simple_fields = true`. It would be nice to do that, so that the downstream app don't have to figure out this detail, and go add the line above to force simple fields in the io broker.

Would it be possible to just expose this macro as a cmake configurable way? I can of course add `-DSTK_USE_SIMPLE_FIELDS` to the compile flags, but that seems hacky...

**Comment by alanw0** [comment]
Yes that's a reasonable request. We'll make that macro configurable via cmake, it will be included in the next stk snapshot into trilinos.
Simplifying stk fields has been a journey, we hope to make simple fields always on before too much longer...

**Comment by alanw0** [comment]
@bartgol this was addressed in #12553
You should now be able to use `-DSTK_USE_SIMPLE_FIELDS` as a cmake argument and it will get defined in the stk configuration file (STK_Trilinos_config.h.in) which should affect all relevant stk source files.

**Comment by bartgol** [comment]
Great, thanks!
